File: aoc15/src/test/java/aoc15/day11/Day11Test.java
package aoc15.day11;

import static org.junit.jupiter.api.Assertions.assertEquals;

import java.util.stream.Stream;
import org.junit.jupiter.params.ParameterizedTest;
import org.junit.jupiter.params.provider.Arguments;
import org.junit.jupiter.params.provider.MethodSource;

class Day11Test {

  @ParameterizedTest
  @MethodSource("newValidPasswordParameters")
  void part1Test(String input, String expectedOutput) {
    var day = new Day11();

    String output = (String) day.part1(input);

    assertEquals(expectedOutput, output);
  }

  private static Stream<Arguments> newValidPasswordParameters() {
    return Stream.of(
        Arguments.of("abcdefgh", "abcdffaa"),
        Arguments.of("ghijklmn", "ghjaabcc")
    );
  }


}

File: aoc20/src/test/java/day11/Day11Test.kt
package day11

import aoc15.loadExample
import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.Test

class Day11Test {

    private var subject: Day11 = Day11()

    private val input = loadExample("day11.txt")

    @Test
    fun part1Examples() {
        assertThat(subject.part1(input))
            .isEqualTo(37)
    }

    @Test
    fun part2Examples() {
        assertThat(subject.part2(input))
            .isEqualTo(26)
    }
}